### PR TITLE
[WIP][FLINK-23654][runtime] Splitting IO and future threadpools and adding advanced configurations

### PR DESCRIPTION
**Pull request by tillrohrmann**
Draft PR for small improvements for #16946.

cc @nicolasraga1, @Aitozi.

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobManagerSharedServices.java [hunk 1]
```diff
@@ -147,10 +164,18 @@ public static JobManagerSharedServices fromConfiguration(
                                 failOnJvmMetaspaceOomError ? fatalErrorHandler : null,
                                 checkClassLoaderLeak));
 
+        final int numberCPUCores = Hardware.getNumberCPUCores();
+        final int numJobManagerFutureThreads =
```
Comment (Aitozi) [inline flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobManagerSharedServices.java:168]: `numJobManagerFutureThreads ` -> `jobManagerFuturePoolSize` ?

File: flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobManagerSharedServices.java [hunk 2]
```diff
@@ -147,10 +164,18 @@ public static JobManagerSharedServices fromConfiguration(
...
+                config.getInteger(JobManagerOptions.JOB_MANAGER_FUTURE_POOL_SIZE, numberCPUCores);
         final ScheduledExecutorService futureExecutor =
                 Executors.newScheduledThreadPool(
-                        Hardware.getNumberCPUCores(),
-                        new ExecutorThreadFactory("jobmanager-future"));
+                        numJobManagerFutureThreads, new ExecutorThreadFactory("jobmanager-future"));
+
+        final int numJobManagerIoThreads =
```
Comment (Aitozi) [inline flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobManagerSharedServices.java:174]: same here

File: flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobManagerSharedServicesTest.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            ScheduledExecutorService futureExecutor = jobManagerSharedServices.getFutureExecutor();
+
+            assertExecutorPoolSize(futureExecutor, CPU_CORES);
+        } finally {
+            jobManagerSharedServices.shutdown();
+        }
+    }
+
+    @Nonnull
+    private JobManagerSharedServices createJobManagerSharedServices(Configuration config)
```
Comment (Aitozi) [inline flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobManagerSharedServicesTest.java:66]: Move these two helper method down to the class ?

**Review by Aitozi — COMMENTED** [review]
LGTM +1 with some minor comments.

**Comment by tillrohrmann** [comment]
#16946 has been merged.

**Pull request closed without merging**